src/debuggability.js

```diff
@@ -161,6 +163,46 @@ Promise.prototype._propagateFrom = function (parent, flags) {
     USE(flags);
 };
 
+Promise.monitor = function () {
+    // Pending promises are stored in array for better performance: number
+    // of simultaneously pending promises is assumed to be relatively small,
```

Comment:
Is that really true? I could imagine `Promise.map` or similar to create a quite huge number of concurrent promises. Adding an `O(n)` step to each of them sounds desastrous.
Wouldn't a ES6 `Set` be a better fit here?